pkg/infra/tower.go

```diff
@@ -23,6 +23,7 @@ type Tower struct {
 	dayInfo        *day.DayInfo
 	healthInfo     *health.HealthInfo
 	mx             sync.RWMutex
+	deadAgents     map[int]int
```

Comment:
I was considering we could have a struct containing al these "states" that we would like to keep track off and log, and then we pass it to tower and sim and base agent. Everyone can then update accordingly the struct and call something like "state.log" which prints the whole state. What do you think ?

Reply:
Actually, that's something we can add on the infra team side when we integrate the frontend logging. I think that should be good for now :)